### PR TITLE
Master pos mobile imp res

### PR DESCRIPTION
**Pull request by res-odoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/point_of_sale/static/src/js/Chrome.js [hunk 1]
```diff
@@ -312,6 +314,9 @@ odoo.define('point_of_sale.Chrome', function(require) {
         _toggleDebugWidget() {
             this.state.debugWidgetIsShown = !this.state.debugWidgetIsShown;
         }
+        _toggleMobileSearchBar(event) {
+            this.state.mobileSearchBarIsShown = event.detail.mobileSearchBarIsShown;
+        }
```
Comment (rfr-odoo) [inline addons/point_of_sale/static/src/js/Chrome.js:321]: if i'm not wrong toggle is "implicit"
is not better to use something like _setMobileSearchBar(event) and in the event set the value ?

File: addons/pos_restaurant/static/src/css/restaurant.css [hunk 2]
```diff
@@ -13,7 +13,7 @@
 
 /* ------ FLOOR SELECTOR ------- */
 
-.floor-selector { 
+.floor-selector {
```
Comment (rfr-odoo) [inline addons/pos_restaurant/static/src/css/restaurant.css:16]: no change on this line

File: addons/pos_restaurant/static/src/xml/ChromeWidgets/BackToFloorButton.xml [hunk 3]
```diff
@@ -4,13 +4,13 @@
...
-            <span> </span>
+            <span t-if="!env.isMobile"> </span>
+            <span t-if="!env.isMobile" t-esc="floor.name" />
+            <span t-if="!env.isMobile"> </span>
             <span class="table-name">
-                <span>( </span>
+                <span t-if="!env.isMobile">( </span>
                 <span t-esc="table.name" />
-                <span> )</span>
+                <span t-if="!env.isMobile"> )</span>
```
Comment (rfr-odoo) [inline addons/pos_restaurant/static/src/xml/ChromeWidgets/BackToFloorButton.xml:13]:
```xml
<t t-if="!env.isMobile">
            <span> </span>
            <span t-esc="floor.name" />
            <span> </span>
</t>
```

File: addons/point_of_sale/static/src/css/pos.css [hunk 4]
```diff
@@ -1548,9 +1548,14 @@ td {
...
+.payment-screen .payment-buttons .button.highlight {
+    background: #e2e2e2 !important;
+    border-color: rgb(200,200,200) !important;
+    color: #555555 !important;
+}
 .payment-screen .payment-buttons .button.highlight .fa {
-    border-color: rgba(0, 0, 0, 0.109804);
-    background: rgba(0, 0, 0, 0.0980392);
+    background: #6EC89B !important;
+    border: solid 1px #64AF8A !important;
```
Comment (rfr-odoo) [inline addons/point_of_sale/static/src/css/pos.css:1558]: try to remove !important and move this section (`.payment-screen .payment-buttons .button.highlight .fa`) after the next section (`.payment-screen .payment-buttons .button .fa`)

File: addons/point_of_sale/static/src/xml/Screens/ProductScreen/ProductScreen.xml [hunk 5]
```diff
@@ -8,16 +8,10 @@
                     <OrderWidget/>
                     <div class="pads">
                         <div class="control-buttons">
-                            <t t-if="env.isMobile">
-                                <t t-if="state.displayControlButtons"
-                                   t-foreach="controlButtons" t-as="cb" t-key="cb.name">
-                                    <t t-if="!_isQuickAccessControlButton(cb.name)" t-component="cb.component" t-key="cb.name"/>
-                                </t>
```
Comment (rfr-odoo) [inline addons/point_of_sale/static/src/xml/Screens/ProductScreen/ProductScreen.xml]: If there is no `firstControlButton` and no `secondControlButton` set in the config nothing than the `_displayAllControlPopup` will be show.

File: addons/pos_restaurant/static/src/js/Screens/FloorScreen/FloorScreen.js [hunk 6]
```diff
@@ -68,6 +69,17 @@ odoo.define('pos_restaurant.FloorScreen', function (require) {
...
+            if (table.order_count !== undefined) {
+                return table.order_count;
+            }
+            return this.env.pos
+                .get_table_orders(table)
+                .filter(o => o.orderlines.length !== 0 || o.paymentlines.length !== 0).length;
+        }
+        get styleButtonBarHandler() {
+            return this.state.view === 'floor-map' ? `background:${this.state.floorBackground}` : '';
+        }
```
Comment (rfr-odoo) [inline addons/pos_restaurant/static/src/js/Screens/FloorScreen/FloorScreen.js:141]: `styleButtonBarHandler()`seen not be used

File: addons/pos_restaurant/static/src/js/Screens/FloorScreen/FloorScreen.js [hunk 7]
```diff
@@ -77,6 +78,33 @@ odoo.define('pos_restaurant.FloorScreen', function (require) {
...
         }
+        _onPinchMove(ev) {
+            const touches = ev.touches;
+            // If two pointers are down, check for pinch gestures
+            if (touches.length === 2) {
+                // Calculate the current distance between the 2 fingers
+                const deltaX = touches[0].pageX - touches[1].pageX;
+                const deltaY = touches[0].pageY - touches[1].pageY;
+                const curDiff = Math.hypot(deltaX, deltaY);
+                if (this.state.pinchLastDeltaDiff === null) {
```
Comment (rfr-odoo) [inline addons/pos_restaurant/static/src/js/Screens/FloorScreen/FloorScreen.js:99]: the scale may be weird as `this.state.pinchLastDeltaDiff` is set only once at the first `_onPinchMove()`

File: addons/point_of_sale/static/src/xml/Screens/TicketScreen/TicketScreen.xml [hunk 8]
```diff
@@ -6,11 +6,21 @@
             <div class="screen-content">
                 <div class="controls">
                     <div class="buttons">
-                        <button t-if="showNewTicketButton" class="highlight" t-on-click="createNewOrder">New Order</button>
-                        <button class="discard" t-on-click="trigger('close-screen')">Discard</button>
+                        <t t-if="env.isMobile">
```
Comment (rfr-odoo) [inline addons/point_of_sale/static/src/xml/Screens/TicketScreen/TicketScreen.xml:9]: To difficult read for me
Proposal (but need to verify the condition)
```xml
<div class="buttons">
    <t t-if="!state.showSearchBar">
        <button t-if="env.isMobile" class="search" t-on-click="toggleSearchBar"><span class="search-icon"><i class="fa fa-search"/></span>Search</button>
        <button t-if="showNewTicketButton" class="highlight" t-on-click="createNewOrder">New Order</button>
        <button class="discard" t-on-click="trigger('close-screen')">Back</button>
    </t>
    <t t-else="">
        <button class="arrow-left" t-on-click="toggleSearchBar"><span class="search-icon"><i class="fa fa-arrow-left"/></span></button>
    </t>
</div>
<t t-if="state.showSearchBar">
    <t t-set="placeholder">Search Tickets...</t>
    <SearchBar config="searchBarConfig" placeholder="placeholder" />
</t>

```

File: addons/point_of_sale/static/src/xml/Screens/TicketScreen/TicketScreen.xml [hunk 9]
```diff
@@ -6,11 +6,21 @@
...
+                            </button>
+                        </t>
+                        <t t-if="!env.isMobile || !state.showSearchBar">
+                            <button t-if="showNewTicketButton" class="highlight" t-on-click="createNewOrder">New Order</button>
+                            <button class="discard" t-on-click="trigger('close-screen')">Back</button>
+                        </t>
                     </div>
                     <t t-set="placeholder">Search Tickets...</t>
-                    <SearchBar config="searchBarConfig" placeholder="placeholder" />
+                    <SearchBar t-if="state.showSearchBar" config="searchBarConfig" placeholder="placeholder" />
```
Comment (rfr-odoo) [inline addons/point_of_sale/static/src/xml/Screens/TicketScreen/TicketScreen.xml:23]: To difficult to read for me
Proposal (but need to verify the condition)
```xml
<div class="buttons">
    <t t-if="!state.showSearchBar">
        <button t-if="env.isMobile" class="search" t-on-click="toggleSearchBar"><span class="search-icon"><i class="fa fa-search"/></span>Search</button>
        <button t-if="showNewTicketButton" class="highlight" t-on-click="createNewOrder">New Order</button>
        <button class="discard" t-on-click="trigger('close-screen')">Back</button>
    </t>
    <t t-else="">
        <button class="arrow-left" t-on-click="toggleSearchBar"><span class="search-icon"><i class="fa fa-arrow-left"/></span></button>
    </t>
</div>
<t t-if="state.showSearchBar">
    <t t-set="placeholder">Search Tickets...</t>
    <SearchBar config="searchBarConfig" placeholder="placeholder" />
</t>

```

File: addons/point_of_sale/static/src/xml/ChromeWidgets/CashierName.xml [hunk 10]
```diff
@@ -3,7 +3,8 @@
 
     <t t-name="CashierName" owl="1">
         <div class="oe_status">
-            <span class="username">
+            <i t-if="env.isMobile" class="fa fa-user" role="img" />
```
Comment (adr-odoo) [inline addons/point_of_sale/static/src/xml/ChromeWidgets/CashierName.xml:6]: Use avatar or title

File: addons/point_of_sale/static/src/xml/Screens/ProductScreen/ProductsWidgetControlPanel.xml [hunk 11]
```diff
@@ -37,7 +37,18 @@
                 </t>
             </t>
             <Portal target="'.pos .search-bar-portal'">
-                <div class="search-box">
+                <div class="search-box" t-if="env.isMobile">
```
Comment (adr-odoo) [inline addons/point_of_sale/static/src/xml/Screens/ProductScreen/ProductsWidgetControlPanel.xml:38]: search-box not aligned with others icons

File: addons/point_of_sale/static/src/js/Screens/ProductScreen/ProductsWidgetControlPanel.js [hunk 12]
```diff
@@ -11,6 +11,9 @@ odoo.define('point_of_sale.ProductsWidgetControlPanel', function(require) {
             super(...arguments);
             this.searchWordInput = useRef('search-word-input');
             this.updateSearch = debounce(this.updateSearch, 100);
+            this.state = useState({
+                mobileSearchBarIsShown: false,
```
Comment (adr-odoo) [inline addons/point_of_sale/static/src/js/Screens/ProductScreen/ProductsWidgetControlPanel.js:15]: this state is already present in chrome.js

File: addons/point_of_sale/static/src/js/Screens/ProductScreen/ProductsWidgetControlPanel.js [hunk 13]
```diff
@@ -24,6 +27,17 @@ odoo.define('point_of_sale.ProductsWidgetControlPanel', function(require) {
                 this.trigger('try-add-product', { searchWordInput: this.searchWordInput });
             }
         }
+        _toggleMobileSearchbar() {
+            this.state.mobileSearchBarIsShown = !this.state.mobileSearchBarIsShown;
+            this.trigger('toggle-mobile-searchbar', {mobileSearchBarIsShown: this.state.mobileSearchBarIsShown});
+            // wait the next render of owl to allow focus item if exist
```
Comment (adr-odoo) [inline addons/point_of_sale/static/src/js/Screens/ProductScreen/ProductsWidgetControlPanel.js:33]: on monted()? We will try to find an alternative together.

File: addons/point_of_sale/static/src/xml/Screens/TicketScreen/TicketScreen.xml [hunk 14]
```diff
@@ -6,11 +6,21 @@
             <div class="screen-content">
                 <div class="controls">
                     <div class="buttons">
-                        <button t-if="showNewTicketButton" class="highlight" t-on-click="createNewOrder">New Order</button>
-                        <button class="discard" t-on-click="trigger('close-screen')">Discard</button>
+                        <t t-if="env.isMobile">
+                            <button t-att-class="{'search': !state.showSearchBar,'arrow-left': state.showSearchBar}" t-on-click="toggleSearchBar">
+                                <span class="search-icon"><i class="fa" t-att-class="{'fa-search': !state.showSearchBar,'fa-arrow-left': state.showSearchBar}"/></span>
+                                <t t-if="!state.showSearchBar">
+                                    Search
```
Comment (adr-odoo) [inline addons/point_of_sale/static/src/xml/Screens/TicketScreen/TicketScreen.xml:13]: Not sure that we want this button on bottom... On other screens like products page, the search bar is on the top.

File: addons/point_of_sale/static/src/css/pos.css [hunk 15]
```diff
@@ -3613,6 +3613,10 @@ td {
 
 @media screen and (max-width: 768px) {
 
+    .pos .receipt-screen .default-view .actions .send-email input {
+        width: 0;
```
Comment (adr-odoo) [inline addons/point_of_sale/static/src/css/pos.css:3633]: drop this commit, not needed anymore

File: addons/pos_restaurant/static/src/xml/ChromeWidgets/BackToFloorButton.xml [hunk 16]
```diff
@@ -4,13 +4,13 @@
     <t t-name="BackToFloorButton" owl="1">
         <span t-if="hasTable" class="order-button floor-button" t-on-click="backToFloorScreen">
             <i class="fa fa-angle-double-left" role="img" aria-label="Back to floor" title="Back to floor" />
-            <span> </span>
-            <span t-esc="floor.name" />
-            <span> </span>
+            <span t-if="!env.isMobile"> </span>
+            <span t-if="!env.isMobile" t-esc="floor.name" />
```
Comment (adr-odoo) [inline addons/pos_restaurant/static/src/xml/ChromeWidgets/BackToFloorButton.xml:8]: Needed on mobile because we can have the same table number

Comment (adr-odoo) [reply addons/pos_restaurant/static/src/xml/ChromeWidgets/BackToFloorButton.xml:8]: Or change demo data, or unique number for tables? to check with POS.

File: addons/point_of_sale/static/src/xml/ChromeWidgets/HeaderButton.xml [hunk 17]
```diff
@@ -4,7 +4,8 @@
     <t t-name="HeaderButton" owl="1">
         <div class="header-button close_button" t-att-class="{ confirm: state.label === 'Confirm' }"
              t-on-click="onClick">
-            <t t-esc="translatedLabel" />
+            <i class="fa fa-sign-out" role="img"/>
```
Comment (adr-odoo) [inline addons/point_of_sale/static/src/xml/ChromeWidgets/HeaderButton.xml:7]: should be visible on home screen only, no?

File: addons/point_of_sale/static/src/css/pos.css [hunk 18]
```diff
@@ -3669,4 +3669,8 @@ td {
     .pos .ticket-screen .pos-search-bar .search input {
         width: 100px;
     }
+
+    .pos .actionpad .button.pay {
+        height: 108px;
```
Comment (adr-odoo) [inline addons/point_of_sale/static/src/css/pos.css:3687]: missing comment

File: addons/pos_restaurant/static/src/js/Screens/FloorScreen/TableSelectWidget.js [hunk 19]
```diff
@@ -0,0 +1,33 @@
+odoo.define('pos_restaurant.TableSelectWidget', function(require) {
```
Comment (adr-odoo) [inline addons/pos_restaurant/static/src/js/Screens/FloorScreen/TableSelectWidget.js:1]: drop this widget

File: addons/point_of_sale/static/src/css/pos.css [hunk 20]
```diff
@@ -3733,4 +3733,9 @@ td {
     .pos .ticket-screen .orders .col.end {
         text-align: end;
     }
+
+    .pos .floor-screen .floor-map {
```
Comment (adr-odoo) [inline addons/point_of_sale/static/src/css/pos.css:3747]: drop this commit

File: addons/point_of_sale/static/src/css/pos.css [hunk 21]
```diff
@@ -3733,4 +3733,25 @@ td {
...
+    }
+
+    .pos .ticket-screen .screen-content,
+    .pos .order-management-screen .flex-container {
+        flex-direction: column-reverse;
+    }
+
+    .pos .ticket-screen .pos-search-bar .filter .options {
+        top: inherit;
+        bottom: calc(100% + 5px);
```
Comment (adr-odoo) [inline addons/point_of_sale/static/src/css/pos.css:3765]: comment

File: addons/point_of_sale/static/src/css/pos.css [hunk 22]
```diff
@@ -3622,6 +3622,10 @@ td {
 
 @media screen and (max-width: 768px) {
 
+    .pos .payment-screen .switchpane {
+        height: initial;
```
Comment (adr-odoo) [inline addons/point_of_sale/static/src/css/pos.css:3629]: No, maybe check if we can remove .switchpane height 100px.

File: addons/point_of_sale/static/src/js/Screens/ProductScreen/ProductScreen.js [hunk 23]
```diff
@@ -35,6 +35,7 @@ odoo.define('point_of_sale.ProductScreen', function(require) {
                 useWithBarcode: true,
             });
             this.state = useState({
+                displayMobileControlButtons: false,
```
Comment (adr-odoo) [inline addons/point_of_sale/static/src/js/Screens/ProductScreen/ProductScreen.js:38]: seems louche

File: addons/point_of_sale/views/pos_config_view.xml [hunk 24]
```diff
@@ -172,6 +172,25 @@
                                 </div>
                             </div>
                         </div>
+                        <div class="col-12 col-lg-6 o_setting_box" >
+                            <div class="o_setting_left_pane"></div>
```
Comment (adr-odoo) [inline addons/point_of_sale/views/pos_config_view.xml:176]: not needed

File: addons/point_of_sale/views/pos_config_view.xml [hunk 25]
```diff
@@ -172,6 +172,25 @@
...
+                            <div class="o_setting_right_pane">
+                                <div class="row ml0">
```
Comment (adr-odoo) [inline addons/point_of_sale/views/pos_config_view.xml:178]: Could we replace all rows div by "content-group" ?

File: addons/point_of_sale/static/src/xml/Screens/ClientListScreen/ClientListScreen.xml [hunk 26]
```diff
@@ -5,6 +5,9 @@
         <div class="clientlist-screen screen" t-on-activate-edit-mode="activateEditMode">
             <div class="screen-content">
                 <div class="top-content">
+                    <div t-if="env.isMobile" class="button back" t-on-click="back">
+                        <i class="fa fa-arrow-left"/>
```
Comment (adr-odoo) [inline addons/point_of_sale/static/src/xml/Screens/ClientListScreen/ClientListScreen.xml:9]: In my mind, I will be better to also have icons on desktop.

File: addons/point_of_sale/static/src/js/Screens/ReceiptScreen/ReceiptScreen.js [hunk 27]
```diff
@@ -27,7 +27,12 @@ odoo.define('point_of_sale.ReceiptScreen', function (require) {
                 // We are doing this because we want the receipt screen to be
                 // displayed regardless of what happen to the handleAutoPrint
                 // call.
-                setTimeout(async () => await this.handleAutoPrint(), 0);
+                setTimeout(async () => {
+                    await this.handleAutoPrint();
+                    if (this.env.isMobile) {
+                        this.el.querySelector('.actions').scrollIntoView();
```
Comment (adr-odoo) [inline addons/point_of_sale/static/src/js/Screens/ReceiptScreen/ReceiptScreen.js:33]: what is the root cause before doing this

File: addons/point_of_sale/static/src/js/Popups/ControlButtonPopup.js [hunk 28]
```diff
@@ -0,0 +1,26 @@
...
+         * @param {string} props.startingValue
+         */
+        constructor() {
+            super(...arguments);
+            this.controlButtons = this.props.controlButtons;
+        }
+    }
+    ControlButtonPopup.template = 'ControlButtonPopup';
+    ControlButtonPopup.defaultProps = {
+        cancelText: 'Back',
```
Comment (adr-odoo) [inline addons/point_of_sale/static/src/js/Popups/ControlButtonPopup.js:19]: What about translations?

File: addons/point_of_sale/static/src/css/pos.css [hunk 29]
```diff
@@ -3707,6 +3707,9 @@ td {
     .pos .ticket-screen .order-table .order-row.delete-button {
         background: #bfbfbf;
     }
+    .pos .ticket-screen .order-table .order-row.delete-button .fa-trash {
+        padding-right: 5px;
```
Comment (adr-odoo) [inline addons/point_of_sale/static/src/css/pos.css:3712]: I prefer margin in this case

File: addons/point_of_sale/static/src/css/pos.css [hunk 30]
```diff
@@ -3772,4 +3772,9 @@ td {
     .pos .popups .modal-dialog .popup-control-buttons .control-buttons .control-button {
         width: 100%;
     }
+
+    .pos .pos-topheader .order-management {
```
Comment (adr-odoo) [inline addons/point_of_sale/static/src/css/pos.css:3777]: could we have the same rule for .order-management and ticket-button?

File: addons/point_of_sale/static/src/xml/Screens/TicketScreen/TicketScreen.xml [hunk 31]
```diff
@@ -6,13 +6,105 @@
...
+                                <t t-if="!env.isMobile">
+                                    Back
+                                </t>
+                            </button>
+                            <button t-if="showNewTicketButton" class="highlight" t-on-click="createNewOrder">New Order</button>
+                        </t>
+                        <t t-if="env.isMobile">
+                            <t t-if="state.showSearchBar">
+                                <button class="arrow-left" t-on-click="state.showSearchBar = !state.showSearchBar;">
+                                    <span class="search-icon">
```
Comment (adr-odoo) [inline addons/point_of_sale/static/src/xml/Screens/TicketScreen/TicketScreen.xml:23]: hide search icon when shear bar is open on mobile

File: addons/point_of_sale/static/src/css/pos.css [hunk 32]
```diff
@@ -3733,4 +3733,25 @@ td {
     .pos .ticket-screen .orders .col.end {
         text-align: end;
     }
+
+    .pos .rightpane-header {
+        flex-wrap: nowrap;
+        overflow-x: auto;
+        overflow-y: hidden;
```
[the remark this answers is not in the document]
Comment (adr-odoo) [reply addons/point_of_sale/static/src/css/pos.css:3751]: to check with "Category Pictures" settings

File: addons/point_of_sale/static/src/xml/ChromeWidgets/CashierName.xml [hunk 33]
```diff
@@ -3,9 +3,19 @@
 
     <t t-name="CashierName" owl="1">
         <div class="oe_status">
-            <span class="username">
-                <t t-esc="username" />
-            </span>
+            <t t-if="env.isMobile">
+                <t t-if="avatar.length">
+                    <img t-att-src="avatar" t-att-alt="username" width="25px" />
```
Comment (adr-odoo) [inline addons/point_of_sale/static/src/xml/ChromeWidgets/CashierName.xml:8]: border-radius: 50%

File: addons/point_of_sale/static/src/xml/ChromeWidgets/CashierName.xml [hunk 34]
```diff
@@ -3,9 +3,19 @@
...
+                </t>
+                <t t-else="">
```
Comment (adr-odoo) [inline addons/point_of_sale/static/src/xml/ChromeWidgets/CashierName.xml:10]: not possible

File: addons/point_of_sale/static/src/js/ChromeWidgets/CashierName.js [hunk 35]
```diff
@@ -14,6 +14,14 @@ odoo.define('point_of_sale.CashierName', function(require) {
                 return '';
             }
         }
+        get avatar() {
+            const cashier = this.env.pos.get_cashier();
+            if (cashier) {
+                return `/web/image/res.users/${cashier.user_id[0]}/avatar_128`;
+            } else {
```
Comment (adr-odoo) [inline addons/point_of_sale/static/src/js/ChromeWidgets/CashierName.js:21]: this.db.get_cashier() || this.get('cashier') || this.employee, '' seems not possible..

File: addons/point_of_sale/static/src/css/pos.css [hunk 36]
```diff
@@ -506,22 +519,13 @@ td {
...
-    float:right;
-    color: rgba(255,255,255,0.4);
-    padding: 14px;
-    line-height: 20px;
-    font-size: 20px;
-    vertical-align:middle;
-    font-style: italic;
     cursor:pointer;
 }
+
```
Comment (adr-odoo) [inline addons/point_of_sale/static/src/css/pos.css:524]: removes this change

File: addons/point_of_sale/static/src/css/pos.css [hunk 37]
```diff
@@ -385,7 +385,25 @@ td {
...
+    align-items: center;
+    padding: 0 10px; /*$dropdown-item-padding-x * .5*/
+}
+
+.pos .status-buttons i.fa {
+    font-size: large;
+}
+
+.pos .status-buttons .avatar {
+    display: inline-block;
```
Comment (adr-odoo) [inline addons/point_of_sale/static/src/css/pos.css:406]: check if needed

File: addons/point_of_sale/static/src/css/pos.css [hunk 38]
```diff
@@ -385,7 +385,25 @@ td {
...
+}
+
+.pos .status-buttons > div {
+    display: flex;
+    align-items: center;
+    padding: 0 10px; /*$dropdown-item-padding-x * .5*/
+}
+
+.pos .status-buttons i.fa {
+    font-size: large;
```
Comment (adr-odoo) [inline addons/point_of_sale/static/src/css/pos.css:402]: Maybe delete font-size: large on header-button ?

**Comment by C3POdoo** [comment]
Dear @res-odoo,

Thank you for your contribution but we are closing it due to inactivity.
We only support the last 3 stable versions so no longer accepts patches into this branch.

We apology if we could not look at your request in time.
If the contribution still makes sense, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._

**Pull request closed without merging**